src/detect-dns-rcode.c

```diff
@@ -0,0 +1,86 @@
+/* Copyright (C) 2019 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
```

Comment:
You can also add your info as author, if you want :)

Reply:
Ooo that would be cool to see when my branch gets merged. Right now my code is all sorts of messed up lol. I'm relearning bitwise operations from back when I studied them at uni

Reply:
💪🏽 
Happens a lot, for these things that we don't use much, after learning.